tests/unit/test_config.py

```diff
@@ -351,21 +351,114 @@ def test_CustomEnum_equalsOp(self):
 		self.assertTrue(defaultNever == CustomEnum.NEVER)
 
 
-class Config_profileUpgradeSteps_upgradeConfigFrom_8_to_9_showMessages(unittest.TestCase):
+def _loadProfile(configString):
+	fn = io.StringIO(configString)
+	profile = configobj.ConfigObj(fn, indent_type="\t", encoding="UTF-8", file_error=False)
+	# Python converts \r\n to \n when reading files in Windows, so ConfigObj can't determine
+	# the true line ending.
+	profile.newlines = "\r\n"
+	return profile
+
+
+class Config_profileUpgradeSteps__upgradeConfigFrom_8_to_9_lineIndent(unittest.TestCase):
+
+	def test_DefaultProfile_Unmodified(self):
+		"""Document formatting Line indentation reporting option not modified in default profile."""
+		
+		configString = "[documentFormatting]"
+		profile = _loadProfile(configString)
+		_upgradeConfigFrom_8_to_9_lineIndent(profile)
+		with self.assertRaises(KeyError):
+			profile['documentFormatting']['reportLineIndentation']
+		with self.assertRaises(KeyError):
+			profile['documentFormatting']['reportLineIndentationWithTones']
+
+	def test_DefaultProfile_LineIndentationExplicitelyOff(self):
+		"""Document formatting Line indentation reporting option explicitely set to off in default profile
+		after having previously selected speech and tones.
+		"""
+		
+		configString = """
+[documentFormatting]
+	reportLineIndentation = False
+	reportLineIndentationWithTones = False
+"""
+		profile = _loadProfile(configString)
+		_upgradeConfigFrom_8_to_9_lineIndent(profile)
+		self.assertEqual(profile['documentFormatting']['reportLineIndentation'], 0)

```

Comment:
An int enum for 0, 1, 2 would be helpful.

Reply:
Do you mean using `config.configFlags.ReportLineIndentation.OFF.value` instead?

I was unsure if it was a good idea to use another part of NVDA's code to unit test this function.

If it's OK for you, I'll convert all the numerical values the same way.

Reply:
Yes, this is ok.

Reply:
Fixed in 559460c89c28be31100a623eac9c4a83f4741b2c. You can now review this PR again.